src/protocol/response_writer.rs

```diff
@@ -228,6 +229,34 @@ impl<'a, C: Connection + 'a> ResponseWriter<'a, C> {
         Ok(())
     }
 
+    /// Write a number as a decimal string, converting every digit to an ascii char.
+    #[allow(clippy::just_underscores_and_digits)]
+    pub fn write_dec<D: PrimInt + CheckedRem>(
+        &mut self,
+        mut digit: D,
+    ) -> Result<(), Error<C::Error>> {
+        if digit.is_zero() {
+            return self.write(b'0');
+        }
+
+        let _1: D = one();
+        let _10 = (_1 << 3) + (_1 << 1);
+        let mut d = digit;
+        let mut pow_10 = _1;
+        // Get the number of digits in digit
+        while d >= _10 {
+            d = d / _10;
+            pow_10 = pow_10 * _10;
+        }
+        // Write every digit from left to right as an ascii char
+        while !pow_10.is_zero() {
+            self.write(b'0' + (digit / pow_10).to_u8().unwrap())?;
```

Comment:
since the generic is down to a single digit here I changed this to extract the 4 LSBs directly, the unwrap is gone now but `write_dec` got a bit lenghty

Reply:
I might end up taking a crack at writing an implementation myself at some point, but at first blush, what you've got now seems fine. I know how annoying it can be doing generic math in Rust, so having this be a wordy function might just be how it is ¯\\\_(ツ)_/¯